### PR TITLE
fix: dragPreview / dropIndicator / dragEndPos viewport offsets

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
before:

https://github.com/toeverything/blocksuite/assets/54364088/12633a02-cbf2-410b-935e-a94a60548944

after:

https://github.com/toeverything/blocksuite/assets/54364088/22db2e16-a9db-43c4-9dc5-4934594783ec

### CODE REVIEW

File: packages/blocks/src/page-block/edgeless/edgeless-page-block.ts [hunk 1]
```diff
@@ -90,6 +90,17 @@ import {
 import { xywhArrayToObject } from './utils/convert.js';
 import { getCursorMode, isCanvasElement, isFrameBlock } from './utils/query.js';
 
+export interface EdgelessViewport {
```
Comment (Flrande) [inline packages/blocks/src/page-block/edgeless/edgeless-page-block.ts:93]: We already have a `PageViewport` interface, and I think `EdgelessPageBlockComponent` and `DocPageBlockComponent` can share the same interface.

https://github.com/toeverything/blocksuite/blob/ee3a4ff482d2459a58cdfa2bb752d055e3774f05/packages/blocks/src/page-block/doc/doc-page-block.ts#L24-L33

File: packages/blocks/src/page-block/edgeless/edgeless-page-block.ts [hunk 2]
```diff
@@ -229,6 +240,41 @@ export class EdgelessPageBlockComponent extends BlockElement<
     return this._viewportElement;
   }
 
+  get viewport(): EdgelessViewport {
+    if (!this.viewportElement) {
```
Comment (Flrande) [inline packages/blocks/src/page-block/edgeless/edgeless-page-block.ts:244]: If the `viewportElement` does not exist, I think it should throw an error directly, so as to maintain consistency with the implementation of `get viewportElement()`.

File: packages/blocks/src/page-block/edgeless/edgeless-page-block.ts (hunk 1 above)
Comment (AyushAgrawal-A2) [reply packages/blocks/src/page-block/edgeless/edgeless-page-block.ts:94]: @Flrande Done

File: packages/blocks/src/page-block/edgeless/edgeless-page-block.ts (hunk 2 above)
Comment (AyushAgrawal-A2) [reply packages/blocks/src/page-block/edgeless/edgeless-page-block.ts]: @Flrande Done
thanks for the quick review

**Review by doodlewind — APPROVED** [review]
Awesome fix!

**Pull request merged**